akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/BasicPersistentBehaviorCompileOnly.scala

```diff
@@ -207,11 +226,11 @@ object BasicPersistentBehaviorCompileOnly {
 
   //#snapshotAndEventDeletes
 
-  val snapshotAndEventsRetention = EventSourcedBehavior[Command, Event, State](
+  EventSourcedBehavior[Command, Event, State](
     persistenceId = PersistenceId("abc"),
     emptyState = State(),
-    commandHandler = (state, cmd) => Effect.noReply, // do something based on a particular command and state
-    eventHandler = (state, evt) => state) // do something based on a particular event and state
+    commandHandler = (state, cmd) => throw new RuntimeException("TODO: process the command & return an Effect"),
+    eventHandler = (state, evt) => throw new RuntimeException("TODO: process the event return the next state"))
```

Comment:
I just wonder if all of these TODO throwing of `RuntimeException` might be clearer
as the `scala.NotImplementedError("TODO: process the command & return an Effect")`. Though I guess the argument is - this is java.

Reply:
ok, changed to `NotImplementedError` for Scala,
the reason for not using `???` is to explain more what is supposed to be done